### PR TITLE
CQ-4348632 Fixed Boolean enum type not working in Dropdown

### PR DESCRIPTION
**Pull request by ravise5**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/dropdown/v1/dropdown/dropdown.html [hunk 1]
```diff
@@ -42,10 +42,9 @@
             data-sly-attribute.multiple="${dropdown.isMultiSelect ? 'multiple' : ''}">
         <option data-sly-test="${dropdown.placeHolder != null}" value="" disabled selected>${dropdown.placeHolder}</option>
         <div data-sly-list.enumName="${dropdown.enumNames}" data-sly-unwrap>
-            <div data-sly-test.enum="${dropdown.enums[enumNameList.index]}" data-sly-unwrap>
-                <option value="${enum}"
+                <option value="${dropdown.enums[enumNameList.index].toString}"
```
Comment (nit23uec) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/dropdown/v1/dropdown/dropdown.html:45]: can `dropdown.enums[enumNameList.index]` be null?

Comment (nit23uec) [reply ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/dropdown/v1/dropdown/dropdown.html:45]: I think we should keep the `data-sly-test ` to account for null (just not assign it to any variable)

Comment (ravise5) [reply ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/dropdown/v1/dropdown/dropdown.html:45]: `dropdown.enums[enumNameList.index]`  is null only when we allow the user to skip either the save value or display value in the edit dialog. 
 Checks for which ig needs to be in place. 
 Currently radiobutton and checkbox also uses this same approach

Comment (ravise5) [reply ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/dropdown/v1/dropdown/dropdown.html:45]: Maybe in a prefill use-case ig it is possible to have null values, but then again we are not doing this check currently for enumNames.

**Review by nit23uec — APPROVED** [review]

**Pull request merged**